File: week_1/day-2-prefix-sum-hashing/variations_que/min_subarray_len.js
// min_subarray_len.js

function minSubArrayLen(k, nums) {
    let left = 0;
    let sum = 0;
    let ans = Infinity;

    for (let right = 0; right < nums.length; right++) {
        sum += nums[right];
        while (sum >= k) {
            ans = Math.min(ans, right - left + 1);
            sum -= nums[left++];
        }
    }

    return ans === Infinity ? 0 : ans;
}

// Example Input
const nums = [2, 3, 1, 2, 4, 3];
const k = 7;

console.log(minSubArrayLen(k, nums)); // Output: 2 ([4,3])

File: week_1/day-2-prefix-sum-hashing/variations_que/min_subarray_len.py
# min_subarray_len.py

def minSubArrayLen(k, nums):
    left = 0
    curr_sum = 0
    ans = float('inf')

    for right in range(len(nums)):
        curr_sum += nums[right]
        while curr_sum >= k:
            ans = min(ans, right - left + 1)
            curr_sum -= nums[left]
            left += 1

    return 0 if ans == float('inf') else ans

# Example Input
nums = [2, 3, 1, 2, 4, 3]
k = 7

# Function Call
print(minSubArrayLen(k, nums))  # Output: 2 ([4,3])
